### DISCUSSION TITLE
Invoicing and invoice storage

### DISCUSSION THREAD

**Discussion by rafalczyzyk** [Budibase/budibase#11994] [Help]
Hi guys.
I am a beginner and I love budibase already. But for the sake of it, I cannot figure out how to create an invoice registering app.
I know there is invoice approval app, but this is not what I need.

I need a place, where I can see rows with separate invoices (easy). But I need a way to add invoices with selections of my products from the same site. I have products database.
What I cannot do is how to add invoice, that has several rows populated with products, their prices and so on. And of course total price, including VAT etc, auto summed up.
As I see it:
I click a button to add new invoice. A pop up screen comes, on the top we have invoice number autogenerated, then a dropdown to select client from clients database, when we select the client, it has to populate some sort of table with their name, address, tax number etc.
Then we have sort of dropdown to add products. So we search fo the product and click on it, it will populate another table with products, each row contains one product with the price (net, gross etc). And each product must be added in separate line below.
On the bottom I need total price and some sort of text describing my payment terms etc.
The logic of this exceeds my knowledge, please help. Picture is example from web what I am after.
Print this to pdf would also be good.
Thank you!
Ralph
![data_create_invoice_options](https://github.com/Budibase/budibase/assets/17220002/ba6d3c7f-6f92-4f2a-b46b-fe1437140126)

**Comment by rafalczyzyk** [comment]
I have found out the plucking tutorial, but this not solve all of my questions:
https://docs.budibase.com/docs/introduction-to-bindings#tutorial-calculating-the-total-cost-of-selected-items

Also, I do not need from you to write me the full code, just point me to the right direction, eg. for this use this and modify this etc...

Nobody?
R.

**Reply by melohagan** [reply]
Hey @rafalczyzyk 

How did you get on?

Here's some links you may find useful:

- https://docs.budibase.com/docs/cascading-dropdown-filters
- https://docs.budibase.com/docs/filter-by-relationships#tutorial-filter-on-selected-relationship-picker-values

**Comment by jjhhrj** [comment]
![image](https://github.com/Budibase/budibase/assets/164453240/c3f9b950-07b7-4804-a752-2e2ed16096b8)
hello can I ask how you create the print button here, thank you
